Composer/packages/ui-plugins/virtual-assistant-creation/src/components/customizeBotPage.tsx

```diff
@@ -0,0 +1,130 @@
+
+const dropdownStyles: Partial<IDropdownStyles> = {
+  dropdown: { width: 300 },
+};
+
+// -------------------- CustomizeBotPage -------------------- //
+type CustomizeBotPageProps = {
+  onDismiss: () => void;
+} & RouteComponentProps<{}>;
+
+export const CustomizeBotPage: React.FC<CustomizeBotPageProps> = (props) => {
+  const { state, setState } = useContext(AppContext);
+  const onDismiss = props.onDismiss;
+
+  const personalityOptions = (): IDropdownOption[] => {
+    return Object.values(AvailablePersonalities).map((personality) => {
+      return { key: personality, text: formatMessage(personality) };
+    });
+  };
+
+  return (
+    <DialogWrapper
+      isOpen
+      dialogType={DialogTypes.CreateFlow}
+      subText={formatMessage(
+        'Give your bot personality, multi-language capabilities and more. You can edit this later in Bot Settings.'
+      )}
+      title={formatMessage('Customize your assistant')}
+      onDismiss={props.onDismiss}
+    >
+      <Stack tokens={verticalStackTokens}>
+        <Stack.Item>
+          <Label>{formatMessage('Bot Name')}</Label>
+          <TextField
+            css={textFieldStyling}
+            value={formatMessage(state.selectedBotName)}
+            onChange={(event: any, newValue?: string) => {
+              setState({ ...state, selectedBotName: newValue as string });
+            }}
+          />
+        </Stack.Item>
+        <Stack.Item>
+          <Label>{formatMessage('User Input')}</Label>
+          <Stack horizontal tokens={horizontalStackTokens}>
+            <Checkbox
+              checked={state.isTextEnabled}
+              label={formatMessage('Text')}
+              onChange={() => {
+                setState({ ...state, isTextEnabled: !state.isTextEnabled });
+              }}
+            />
+            <Checkbox
+              checked={state.isSpeechEnabled}
+              label={formatMessage('Speech')}
+              onChange={() => {
+                setState({ ...state, isSpeechEnabled: !state.isSpeechEnabled });
+              }}
+            />
+          </Stack>
+        </Stack.Item>
+        <Stack.Item>
+          <Dropdown
+            label={formatMessage('Personality')}
+            options={personalityOptions()}
+            placeholder={formatMessage('Select an option')}
+            selectedKey={state.selectedPersonality}
+            styles={dropdownStyles}
+            onChange={(ev: any, option?: IDropdownOption) => {
+              setState({ ...state, selectedPersonality: AvailablePersonalities[option?.text as string] });
+            }}
+          />
+        </Stack.Item>
+        <Stack.Item>
+          <Label>{formatMessage('Greeting Message')}</Label>
+          <TextField
+            css={textFieldStyling}
+            value={formatMessage(state.selectedGreetingMessage)}
+            onChange={(ev?: any, newValue?: string) => {
+              setState({ ...state, selectedGreetingMessage: newValue as string });
```

Comment:
With this approach, every state change will update AppContext which will re-render every component that uses AppContext.

Reply:
I was wondering if this was the case. I initially had this implemented with redux when I was working on it outside composer and refactored it with hooks when moving it into Composer. I now realize with the refactor that it is calling excessive rerenders, good catch!